kernel/src/debug.rs

```diff
@@ -17,9 +39,36 @@ use core::fmt::{Arguments, Result, Write, write};
 use core::ptr::{read_volatile, write_volatile};
 use core::str;
 use driver::Driver;
+use hil;
 use mem::AppSlice;
 use returncode::ReturnCode;
 
+///////////////////////////////////////////////////////////////////
+// debug_gpio! support
+
+pub static mut DEBUG_GPIOS: (Option<&'static hil::gpio::Pin>,
+ Option<&'static hil::gpio::Pin>,
+ Option<&'static hil::gpio::Pin>) = (None, None, None);
+
+pub unsafe fn assign_gpios(gpio0: Option<&'static hil::gpio::Pin>,
+                                 gpio1: Option<&'static hil::gpio::Pin>,
+                                 gpio2: Option<&'static hil::gpio::Pin>) {
+    DEBUG_GPIOS.0 = gpio0;
+    DEBUG_GPIOS.1 = gpio1;
+    DEBUG_GPIOS.2 = gpio2;
+}
+
+/// In-kernel gpio debugging, accepts any GPIO HIL method
```

Comment:
Does anything actually guarantee that you're using a valid method on the GPIOs? What if I call `debug_gpio!(0, transmit);`?

Reply:
```
error[E0599]: no method named `transmit` found for type `&kernel::hil::gpio::Pin` in the current scope
   --> src/main.rs:188:20
    |
188 |     debug_gpio!(0, transmit);
    |                    ^^^^^^^^
```